sql/promscale--0.0.0.sql

```diff
@@ -505,3 +506,361 @@ CREATE TYPE _prom_ext.GapfillDeltaTransition (
+        --those can be found further down.
+
+        --this code does not need to say "ok" to all types of objects, just those that exists in the connector 0.10.0
+        --schema. On the other hand it /must/ fail all other objects.
+        with recursive cte AS (
+            select
+                    0 as level,
+                    ARRAY[pg_describe_object(classid, objid, objsubid)] descr,
+                false is_cycle,
+                classid,
+                objid,
+                objsubid
+            from pg_depend
+            where deptype='e' and
+                refclassid='pg_catalog.pg_extension'::regclass and
+                refobjid = (select oid from pg_extension where extname='promscale')
+            UNION ALL
+            select
+                level + 1,
+                cte.descr || pg_describe_object(dep.classid, dep.objid, dep.objsubid),
+                pg_describe_object(dep.classid, dep.objid, dep.objsubid) = ANY(descr),
+                dep.classid,
+                dep.objid,
+                dep.objsubid
+            from cte
+            inner join pg_depend dep ON (cte.classid = dep.refclassid
+                                    AND cte.objid = dep.refobjid)
+            where NOT is_cycle
+        ),
+        with_owner as (
+            SELECT
+                CASE
+                WHEN classid = 'pg_class'::regclass THEN
+                    (select relowner from pg_class where oid = objid)
+                WHEN classid = 'pg_type'::regclass THEN
+                    (select typowner from pg_type where oid = objid)
+                WHEN classid = 'pg_proc'::regclass THEN
+                    (select proowner from pg_proc where oid = objid)
+                WHEN classid = 'pg_namespace'::regclass THEN
+                    (select nspowner from pg_namespace where oid = objid)
+                WHEN classid = 'pg_operator'::regclass THEN
+                    (select oprowner from pg_operator where oid = objid)
+                ELSE
+                    -1
+                END as owner,
+                *
+            FROM cte
+            --each of these classes is handled separately below
+            WHERE classid NOT IN (
+                'pg_constraint'::regclass,
+                'pg_trigger'::regclass,
+                'pg_rewrite'::regclass,
+                'pg_attrdef'::regclass
+            )
+        )
+        select array_agg (
+                'Object ' || pg_describe_object(classid, objid, objsubid) || 'is owned by ' || owner::text
+        )
+        into errors
+        from with_owner
+        where (
+            --either the owner is the object owner
+            (owner = obj_owner_oid)
+            OR
+            --or prom_admin (these are for metric and series tables)
+            (owner = (SELECT oid FROM pg_roles WHERE rolname = 'prom_admin') AND classid IN ('pg_class'::regclass, 'pg_type'::regclass))
```

Comment:
why is pg_type considered here? metric and series tables would both be pg_class, right?

Reply:
each pg_class creates a corresponding pg_type internally, always. It's for the record type.